[ #5134 ] remove agda-licenses

andreasabel opened the pull request:
Fix #5134, removing src/agda-licenses.  (See discussion there.)

andreasabel committed:
[ #5134 ] license-report only needs cabal v2-configure

[ci skip]

andreasabel committed:
[ fix #5134 ] remove agda-licenses

Superseded by doc/license-report.

[ci skip]

nad commented:
Some things that could be mentioned in `README.md`:

* I guess you should run `cabal v2-configure` with suitable flags (like `--enable-tests` or `-fenable-cluster-counting`).

* Agda relies on things that are not listed, including the GHC run-time system and zlib.

andreasabel committed:
[ #5134 ] licence-report Makefile: add invocation of cabal configure

[ci skip]

andreasabel commented:
>     * I guess you should run `cabal v2-configure` with suitable flags (like `--enable-tests` or `-fenable-cluster-counting`).

Good point!  I also added `make configure` with a suitable invocation of `cabal configure` (did I add all necessary flags?).

>     * Agda relies on things that are not listed, including the GHC run-time system and zlib.

I see `zlib` listed in the _direct dependencies_ (just before the _indirect transitive dependencies_).

> including the GHC run-time system 

Wouldn't have thought that this has a license. (It is code produced by the compiler from the (almost) empty source file.)
Which license does it have?

nad commented:
> I see `zlib` listed in the _direct dependencies_ (just before the _indirect transitive dependencies_).

That is presumably the Haskell library called `zlib`, not the [underlying library](https://zlib.net/). As another example we make use of the [ICU](http://site.icu-project.org/) library.

> > including the GHC run-time system
> 
> Wouldn't have thought that this has a license. (It is code produced by the compiler from the (almost) empty source file.)
> Which license does it have?

I don't know.

andreasabel commented:
> That is presumably the Haskell library called `zlib`, not the [underlying library](https://zlib.net/). As another example we make use of the [ICU](http://site.icu-project.org/) library.

These are not captured by `cabal-plan`.

So how to proceed?  Do you approve this PR?  If not, do you want to update `agda-licenses` to work with the latest `Cabal`?

nad commented:
> So how to proceed? Do you approve this PR? If not, do you want to update `agda-licenses` to work with the latest `Cabal`?

The `agda-licenses` program handles these problems by warning the user:

  https://github.com/agda/agda/blob/a9e4e1c823c86ee11cfa70b8dbd5160aa3566000/src/agda-licences/Main.hs#L29-L36

Above I suggested that you could add some text to `README.md`.

andreasabel committed:
[ #5134 ] license-report/README: disclaimer from nad's agda-licenses

We depend on some non-Haskell libraries that are accessed through
packages like zlib.

See https://github.com/agda/agda/pull/5195#issuecomment-781527984

[ci skip]

andreasabel commented:
I added the disclaimer to the README.

Thanks @nad, for the comments!  I am going ahead now and merge this.

nad commented on doc/license-report/README.md:
```diff
@@ -4,6 +4,14 @@ Agda dependencies license report
+Note that the binary may depend on non-Cabal packages as well.
+
+The licence printed for the rts package is that of (some version of)
+the ghc package.
```
I seem to recall that the `rts` package is not mentioned in the output of `cabal-plan`.

andreasabel replied:
Fixed in 8c01e3aef641accc59ae822edd4b7a79cc2b765d
